.github/workflows/macos.yaml

```diff
@@ -1,38 +1,41 @@
 name: macos
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+    - 'main'
+  pull_request:
+    branches:
+    - '*'
 
 jobs:
-
   macos-build:
-    runs-on: ${{ matrix.macos-version }}
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        macos-version:
-          - 'macos-latest'
+        os:
+          - macos-13
+          - macos-latest
 
     steps:
       - name: Sync repository
         uses: actions/checkout@v2
 
-      - name: Cache Qt
-        id: cache-qt
-        uses: actions/cache@v1  # not v2!
-        with:
-          path: '${{ github.workspace }}/qt_installation/'
-          key: ${{ runner.os }}-QtCache
+      - name: Prepare Homebrew and Install Dependencies
+        run: |
+          rm -rf /usr/local/var/homebrew/locks
```

Comment:
is this necessary? seems like it goes against homebrew policy

Reply:
brew install is not happy sometimes. 
replaced with 
brew cleanup --prune=0